### PR TITLE
Add autorest option to fix const stuttering

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
To avoid breaking changes, the option is off by default.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@gracewilcox FYI

File: packages/autorest.go/test/databoxedge/armdataboxedge/zz_constants.go [hunk 1]
```diff
@@ -303,6 +257,52 @@ func PossibleDayOfWeekValues() []DayOfWeek {
 	}
 }
 
+// DeviceKind - The kind of the device.
+type DeviceKind string
+
+const (
+	DataBoxEdgeDeviceKindAzureDataBoxGateway    DeviceKind = "AzureDataBoxGateway"
```
Comment (RickWinter) [inline packages/autorest.go/test/databoxedge/armdataboxedge/zz_constants.go:264]: Why isn't this `DeviceKindAzureDataBoxGateway`? The `DataBoxEdge` portion seems unnecessary.

Comment (jhendrixMSFT) [reply packages/autorest.go/test/databoxedge/armdataboxedge/zz_constants.go:264]: Yes good catch, will fix.

Comment (jhendrixMSFT) [reply packages/autorest.go/test/databoxedge/armdataboxedge/zz_constants.go]: Fixed

File: packages/autorest.go/src/transform/namer.ts [hunk 2]
```diff
@@ -100,23 +100,69 @@ export async function namer(session: Session<CodeModel>) {
   model.language.go!.singleClient = singleClient;
 
   // fix up type names
-  const structNames = new Set<string>();
+  // final names are added to typeNames which is used to detect collisions
+  const typeNames = new Set<string>();
```
Comment (chlowell) [inline packages/autorest.go/src/transform/namer.ts:104]: Is this just `names` now that there are consts in it too?

**Review by chlowell — APPROVED** [review]

File: packages/autorest.go/src/transform/namer.ts (hunk 2 above)
Comment (jhendrixMSFT) [reply packages/autorest.go/src/transform/namer.ts:104]: Pretty much, but I felt names was a little too generic. It's really `structAndConstNames` but I went with the shorter `typeNames`.

**Pull request merged**